paimon-core/src/main/java/org/apache/paimon/utils/RowDataPartitionComputer.java

```diff
@@ -52,7 +55,19 @@ public LinkedHashMap<String, String> generatePartValues(InternalRow in) {
 
         for (int i = 0; i < partitionFieldGetters.length; i++) {
             Object field = partitionFieldGetters[i].getFieldOrNull(in);
```

Comment:
Can we introduce a StringFormatter to replace FieldGetter?

Reply:
Good idea